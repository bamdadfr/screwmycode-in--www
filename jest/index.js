export { JestWithStyledComponents } from './jest-with-styled-components'