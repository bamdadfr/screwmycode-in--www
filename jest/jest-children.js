import React from 'react'

/**
 * @function
 * @name JestChildren
 * @description jest: dummy children
 * @returns {React.ReactElement} - react component
 */
export function JestChildren () {

    return (
        <>
            <span>children</span>
            <span>children</span>
            <span>children</span>
        </>
    )

}