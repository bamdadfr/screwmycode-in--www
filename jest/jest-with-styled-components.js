import React from 'react'
import { ThemeProvider } from 'styled-components'
import { Global, Theme } from '../src/app/styles'

/**
 * @function
 * @name JestWithStyledComponents
 * @description jest: wrapper with recoil and styled-components for tests
 * @param {React.ReactElement} Component - react sub component
 * @returns {React.ReactElement} - react component
 */
export function JestWithStyledComponents (Component) {

    return (
        <>
            <Global/>
            <ThemeProvider theme={Theme}>
                {Component}
            </ThemeProvider>
        </>
    )

}