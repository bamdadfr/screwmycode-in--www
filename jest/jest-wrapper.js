import React from 'react'
import { RecoilRoot } from 'recoil'
import { ThemeProvider } from 'styled-components'
import { ThemeStyles } from '../src/styles/theme.styles'

/**
 * @function
 * @name JestWrapper
 * @description jest: wrapper with recoil and styled-components for tests
 * @param {React.ReactNode} Component - react sub component
 * @returns {React.ReactNode} - react component
 */
export function JestWrapper (Component) {

    return (
        <RecoilRoot>
            <ThemeProvider theme={ThemeStyles}>
                {Component}
            </ThemeProvider>
        </RecoilRoot>
    )

}