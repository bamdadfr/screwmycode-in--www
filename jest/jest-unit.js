import React from 'react'
import { shallow } from 'enzyme'
import { JestWithStyledComponents } from './jest-with-styled-components'

/**
 * @function
 * @name JestUnit
 * @description jest: get unit from enzyme
 * @param {React.ReactElement} Component - react component
 * @param {boolean} [wrap=false] - wrap with state and theme...
 * @returns {object} - enzyme shallow object
 */
export function JestUnit (Component, wrap = false) {

    if (wrap) {

        return shallow (
            JestWithStyledComponents (Component),
        )

    }

    return shallow (Component)

}