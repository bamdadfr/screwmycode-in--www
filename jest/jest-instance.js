import renderer from 'react-test-renderer'
import { JestWrapper } from './jest-wrapper'

export const JestInstance = (Component) => renderer.create (JestWrapper (Component)).toJSON ()