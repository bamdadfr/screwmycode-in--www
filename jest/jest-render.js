import { render } from '@testing-library/react'
import { JestWrapper } from './jest-wrapper'

export const JestRender = (Component) => render (JestWrapper (Component))