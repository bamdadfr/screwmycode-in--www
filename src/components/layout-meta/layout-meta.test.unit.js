import React from 'react'
import LayoutMetaComponent from './layout-meta.component'
import { JestInstance } from '../../../jest/jest-instance'

const instance = JestInstance (<LayoutMetaComponent/>)

it ('should mount and match', () => {

    expect (instance).toBeDefined ()

    expect (instance).toMatchSnapshot ()

})
