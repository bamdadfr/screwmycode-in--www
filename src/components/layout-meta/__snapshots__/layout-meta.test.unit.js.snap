// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should mount and match 1`] = `null`;
